### ISSUE TITLE
SOLVED: Scaling on 2.2.4?

### ISSUE THREAD

**Issue by emwhy** [wout/svg.js#432]
I am trying to figure out how scaling works on 2.2.4, which is different than 1.1.0.

On 1.1.0, the scaling is based on point (0, 0).  So objects were scaled with (0, 0) as the anchor point.  This worked well for me, since I could easily predict where the object is going to be after scaling.  If I wanted to scale an object based on the center point, all I had to do was to draw an object, so that point (0, 0) as the center.

But it seems that's not the case with 2.2.4.  I am not even sure where the anchor point is, and this is causing me some issue.  x() and y() value returns different issues depending on my scroll on my browser.

Furthermore, when I attempt to move the object back to (0, 0) point after scaling by calling move(0, 0) method doesn't seem to put the object back to that point.

Can you tell me what's going on with scaling, and how to get around it on 2.2.4?  If it is a bug, I would like to have this fixed, since Chrome is showing some warnings on version 1.1.0, saying that a certain object/method will be deprecated.

Thanks,
Em

ex:
// Each of these objects are group of different shapes.  Scale factor is less than 1 (i.e., 0.125, 0.25, 0.5, 1)
  SVG.get("route_path_layer_group").scale(scale, scale);
  SVG.get("location_layer_group").scale(scale, scale);
  SVG.get("object_layer_group").scale(scale, scale);
  SVG.get("map_group").scale(scale, scale);

// An attempt to put the groups back to where they are supposed to be, although it doesn't seem to work.  I didn't even need these lines in 1.1.0 ... 
  SVG.get("route_path_layer_group").move(0, 0);
  SVG.get("location_layer_group").move(0, 0);
  SVG.get("object_layer_group").move(0, 0);
  SVG.get("map_group").move(0, 0);

// This should return (0, 0), but it doesn't.
  console.log("location layer x = " + SVG.get("location_layer_group").x() + " y = " + SVG.get("location_layer_group").y());

**Comment by Fuzzyma** [comment]
A good starting is to read the documentation which is pretty accurate.
I admit that it lacks the key to this problem. You can pass the origin as parameters to the scale function as you would do with rotations.

```javascript
el.scale(2, 0, 0)
```

You can read here about that: https://github.com/wout/svg.js/#scale-1

Note: its just coincident that the `move()` function when called on a group will result in a translation of the element. However this is not true for all other shapes. So moving something in the center for furtehr transformation whould be done with translate and not with move.
However take a look at the `around()` method in the matrix code which does exactly this

**Comment by emwhy** [comment]
Hi Fuzzyma,

Thanks for the information.  That worked.

I am just updating from 1.1.0 to 2.2.4.  I wasn't aware that there was a new scale parameter.  I had read the documentation very thoroughly back then, but I guess I need to do that again.  This kind of change is hard to track down sometimes, especially when you already have fairly establish code that worked in another version.

The reason I didn't try to update earlier was because it was giving me error whenever I tried ... which I finally got around to figuring out what it was (issue #431).

This finally allows me to move to the latest version.  Again, thank you!
Em
